### PR TITLE
Container type Row added to attribute form

### LINKED ISSUE #8231: Add new attribute form container type "Row" (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#52914
Author: @nyalldawson
QGIS version: 3.32

**Add new attribute form container type "Row"**

### PR Description:
This container always lays out child widgets in a horizontal row, where the number of columns is automatically determined by the number of child widgets.

It's useful for creation of compact forms, where no space will be wasted by assigning extraneous horizontal width to widgets where the expected values will always be short.

Eg: creating 3 rows with 3, 2, 1 child widgets respectively results in the layout:

    Attr 1: [...] Attr 2: [...] Attr 3: [...]
    Attr 4: [..........] Attr 5: [..........]
    Attr 6: [...............................]

Without the option of row containers then the all horizontal rows will have the same number of columns, eg:

    Attr 1: [...] Attr 2: [...] Attr 3: [...]
    Attr 4: [...] Attr 5: [...] Attr 6: [...]

(leaving insufficient horizontal length for attributes 4-6), or

    Attr 1: [..........] Attr 2: [..........]
    Attr 2: [..........] Attr 3: [..........]
    Attr 4: [..........] Attr 5: [..........]
    Attr 6: [..........]

(resulting in wasted horizontal space next to attribute 6, and an extra row taking up vertical space)

Sponsored by NIWA

### Commits tagged with [need-docs] or [FEATURE]

### PR DESCRIPTION
**Pull request by selmaVH1**
fixes #8231 

- [ ] Backport to LTR documentation is requested

<!---

### CODE REVIEW

File: docs/user_manual/working_with_vector/vector_properties.rst [hunk 1]
```diff
@@ -2205,8 +2205,10 @@ directly linked to a particular field (like the HTML/QML widgets or the
    #. Then set the properties of the container, ie:
 
       * the :guilabel:`Label`: the title that will be used for the container
-      * the :guilabel:`Container Type`: it can be a :guilabel:`Tab`
-        or a :guilabel:`Group box in container` (a collapsible group box inside a tab or another group)
+      * the :guilabel:`Container Type`: it can be a :guilabel:`Tab`,
+        :guilabel:`Group box in container` (a collapsible group box inside a tab or another group)
+        or a :guilabel:`Row` (a container type that allows you to arange your widgets in a horizonatl row,
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/vector_properties.rst:2210]:
```suggestion
        or a :guilabel:`Row` (a container type that allows you to arange your widgets in a horizontal row,
```

Comment (DelazJ) [reply docs/user_manual/working_with_vector/vector_properties.rst:2210]: Should we also clearly mention the "Within" option that is available for both group and row types?

File: docs/user_manual/working_with_vector/vector_properties.rst [hunk 2]
```diff
@@ -2205,8 +2205,12 @@ directly linked to a particular field (like the HTML/QML widgets or the
...
 
       * the :guilabel:`Label`: the title that will be used for the container
-      * the :guilabel:`Container Type`: it can be a :guilabel:`Tab`
-        or a :guilabel:`Group box in container` (a collapsible group box inside a tab or another group)
+      * the :guilabel:`Container Type`: it can be a :guilabel:`Tab`,
+        :guilabel:`Group box in container` (a collapsible group box inside a tab or another group)
+        or a :guilabel:`Row` (a container type that allows you to arrange your widgets in a horizontal row,
+        automatically determining the number of columns based on the number of widgets),
+      * the :guilabel:`Within` option: provides additional flexibility for customizing the arrangement
+        of the widgets within each container,
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/vector_properties.rst:2213]: Maybe am I misunderstanding your text, but the "within" option is for selecting (or not) an **existing container** in which you would add the one you are creating. Is that what you mean?

Comment (selmaVH1) [reply docs/user_manual/working_with_vector/vector_properties.rst]: Yes, I meant that, but it does sound confusing, so I changed it. I hope it is clear now.

**Pull request merged**